### PR TITLE
servemodule: allow port configuration with flamingo config

### PR DESCRIPTION
**Pull request by SchiffFlieger**
This change allows to configure the listened port with default flamingo config. I kept the `-a` parameter for the `serve` command for backward compatibility. We can think about removing it in the future, because it's possible to use the `--flamingo-config "core.serve.port: 3322"` parameter instead.

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -349,6 +352,11 @@ func (a *servemodule) Configure(injector *dingo.Injector) {
 	})
 }
 
+// CueConfig for the module
+func (a *servemodule) CueConfig() string {
+	return `core: serve: port: number | *3322`
```
Comment (tessig) [inline app.go:357]:
```suggestion
	return `core: serve: port: int | *3322`
```

Comment (tessig) [reply app.go:357]: also set a constraint for the port range

Comment (SchiffFlieger) [reply app.go]: I added the constraint for valid port range, but it's not possible to declare it as `int`:
```
cue: marshal error at path core.serve.port: conflicting values ((int & >=0) & <=65535) and 1234 (mismatched types int and float)
```

Comment (bastianccm) [reply app.go]: I'll look into it

**Review by bastianccm — APPROVED** [review]

**Pull request merged**